### PR TITLE
[EVNT-742] Fix setup call after update

### PR DESCRIPTION
**Pull request by fellipeh**
Fix setup call after update

### CODE REVIEW

**Review by jpramos123 — APPROVED** [review]
LGTM

File: app/redhat_insights/appserver/static/javascript/views/splunk_helpers.js [hunk 1]
```diff
@@ -1,5 +1,79 @@
...
+
+  // Check for the existence of the configuration file
+  var configuration_file_exist = does_configuration_file_exist(
+    splunk_js_sdk_service_configurations,
+    configuration_file_name
+  );
+
+  // If the configuration file doesn't exist, create it
+  if (!configuration_file_exist) {
+    await create_configuration_file(
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/splunk_helpers.js:28]: Why are we creating the configuration file in a get function?

File: app/redhat_insights/appserver/static/javascript/views/setup_configuration.js [hunk 2]
```diff
@@ -73,7 +78,30 @@ define(["uuid"], function(uuidlib) {
...
+      const service = create_splunk_js_sdk_service(splunk_js_sdk, application_name_space)
+      const configFileName = "app";
+      const stanzaName = "install";
+      
+      const isConfigured = await SplunkHelpers.get_config_value(service, configFileName, stanzaName);
+      if (isConfigured){
+        redirect_to_splunk_app_homepage(app_name);
+      }
+    }catch (error){
+      throw new Error(error);
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/setup_configuration.js:98]: Why are we re-throwing the same error?

File: app/redhat_insights/appserver/static/javascript/views/setup_configuration.js [hunk 3]
```diff
@@ -73,7 +78,30 @@ define(["uuid"], function(uuidlib) {
...
+      sharing: "app",
+    };
+    try{
+      const service = create_splunk_js_sdk_service(splunk_js_sdk, application_name_space)
+      const configFileName = "app";
+      const stanzaName = "install";
+      
+      const isConfigured = await SplunkHelpers.get_config_value(service, configFileName, stanzaName);
+      if (isConfigured){
+        redirect_to_splunk_app_homepage(app_name);
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/setup_configuration.js:95]: Shouldn't this check whether the app is configured? If so, then the redirect is outside of its scope.

File: app/redhat_insights/appserver/static/javascript/views/setup.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 
 define(['./setup_configuration'], function(Config) {
 
+  console.debug("Entering setup.js");
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/setup.js:4]: Left out debug logging.

File: app/redhat_insights/appserver/static/javascript/views/app.js [hunk 5]
```diff
@@ -230,6 +248,8 @@ define(["react", "splunkjs/splunk", "splunkjs/mvc", "app/views/setup"], function
     isSetupOpened, setIsSetupOpened, hecUrl, isUrlCopied,
     setIsUrlCopied, splunkVersion, appVersion
   }) => {
+    console.info('SetupIntegration');
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/app.js:251]: Left out debug logging.

File: app/redhat_insights/appserver/static/javascript/views/app.js [hunk 6]
```diff
@@ -1,7 +1,25 @@
...
   const e = react.createElement;
+  // Initialize a Splunk Javascript SDK Service instance
+  const http = new splunkjs.SplunkWebHttp();
+  const service = new splunkjs.Service(
+      http,
+      appNamespace,
+  );
+
+  // Validate if the app is already configured
+  Config.is_app_configured(splunk_js_sdk);
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/app.js:21]: Please move this and those Splunk services to appropriate React component so it isn't called on import.

**Review by vkrizan — CHANGES_REQUESTED** [review]

File: app/redhat_insights/appserver/static/javascript/views/app.js [hunk 7]
```diff
@@ -1,7 +1,25 @@
...
+  sharing: "app",
+};
+
+define(["react", "splunkjs/splunk", "splunkjs/mvc", "app/views/setup", 'app/views/setup_configuration'], function (react, splunk_js_sdk, mvc, Setup, Config) {
+  console.debug("Entering app.js");
+
   const e = react.createElement;
+  // Initialize a Splunk Javascript SDK Service instance
+  const http = new splunkjs.SplunkWebHttp();
+  const service = new splunkjs.Service(
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/app.js:15]: Are these services used?

File: app/redhat_insights/appserver/static/javascript/views/app.js [hunk 8]
```diff
@@ -1,7 +1,25 @@
...
 
-define(["react", "splunkjs/splunk", "splunkjs/mvc", "app/views/setup"], function (react, splunk_js_sdk, mvc, Setup) {
+export const appNamespace = {
+  owner: "Red Hat",
+  app: "redhat_insights",
+  sharing: "app",
+};
+
+define(["react", "splunkjs/splunk", "splunkjs/mvc", "app/views/setup", 'app/views/setup_configuration'], function (react, splunk_js_sdk, mvc, Setup, Config) {
+  console.debug("Entering app.js");
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/app.js:10]: Left out logging.

File: app/redhat_insights/appserver/static/javascript/views/app.js [hunk 9]
```diff
@@ -1,7 +1,25 @@
...
+
   const e = react.createElement;
+  // Initialize a Splunk Javascript SDK Service instance
+  const http = new splunkjs.SplunkWebHttp();
```
Comment (vkrizan) [inline app/redhat_insights/appserver/static/javascript/views/app.js:14]: `splunkjs` is not defined, or is it?

**Pull request merged**